src/createServer.js

```diff
@@ -1,8 +1,45 @@
 'use strict';
 
+const http = require('http');
+const path = require('path');
+const fsPromises = require('fs').promises;
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer(async (req, res) => {
+    res.setHeader('Content-Type', 'text/plain');
+
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    if (!url.pathname.startsWith('/file')) {
+      res.statusCode = 400;
+      res.statusMessage = 'Path should start with "/file/"';
+      res.end();
+
+      return;
+    }
+
+    const requestedPath = url.pathname.slice(6) || 'index.html';
+
+    if (requestedPath.includes('//')) {
+      res.statusCode = 404;
+      res.statusMessage = 'Invalid file path: double slashes are not allowed.';
+      res.end();
+
+      return;
+    }
+
+    const realPath = path.join(__dirname, '..', 'public', requestedPath);
+
+    try {
+      const file = await fsPromises.readFile(realPath, 'utf8');
+
+      res.statusCode = 200;
+      res.end(file);
+    } catch (error) {
+      res.statusCode = 404;
+      res.end('404 Not found');
```

Comment:
The response body for 404 errors is still generic. Consider providing more detailed information to help users understand the issue, such as specifying the requested path that was not found.